### PR TITLE
Feature ADF-1102 CI pipeline

### PR DESCRIPTION
**Pull request by wazelin**
# [ADF-1102](https://oat-sa.atlassian.net/browse/ADF-1102)

This PR aims to integrate the extension with a new CI pipeline defined by https://github.com/oat-sa/tao-extension-ci-action

⚠️ It is expected that PHP > 7.4 pipeline fail ⚠️

## Changelog
- ci: add a CI action
- fix: avoid _exact_ version dependencies on semver-supporting packages
- fix: initialize `EventManager` in the ontology mock
- fix: `LegacyServiceGatewayTest::testHasWillReturnTrueForConfigurableService()`
- fix: `ClassRepositoryTest`'s mocks configuration

### CODE REVIEW

File: test/unit/core/DependencyInjection/LegacyServiceGatewayTest.php [hunk 1]
```diff
@@ -87,6 +87,6 @@ public function testHasWillReturnTrueForConfigurableService(): void
             ->method('has')
             ->willReturn(false);
 
-        $this->assertTrue($this->subject->has(ControllerService::class));
+        $this->assertFalse($this->subject->has(ControllerService::class));
```
Comment (kilatib) [inline test/unit/core/DependencyInjection/LegacyServiceGatewayTest.php:90]: You change the test meaning completely opposite is it really needed?

File: .github/workflows/continuous-integration.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+jobs:
+  build:
+
+    runs-on: ${{ matrix.operating-system }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        operating-system: [ ubuntu-latest ]
+        php-versions: [ '7.2', '7.3', '7.4', '8.0', '8.1' ]
```
Comment (kilatib) [inline .github/workflows/continuous-integration.yaml:18]: If 8.1 it is optionally requirements and looks like it not pass the tests should we remove it?

File: test/unit/core/resource/Repository/ClassRepositoryTest.php [hunk 3]
```diff
@@ -147,7 +147,7 @@ private function createClass(string $uri = null, string $label = null): core_ker
         $class
             ->expects($uri !== null ? $this->once() : $this->never())
             ->method('getUri')
-            ->willReturn($uri);
+            ->willReturn((string)$uri);
```
Comment (kilatib) [inline test/unit/core/resource/Repository/ClassRepositoryTest.php:150]: Why we need typecast for URI?

Comment (wazelin) [reply test/unit/core/resource/Repository/ClassRepositoryTest.php:150]: It's not nullable in the original method.

File: test/unit/core/resource/Service/ResourceDeleterTest.php [hunk 4]
```diff
@@ -109,17 +109,12 @@ public function testDeleteSuccessWithoutParentClass(): void
...
-            ->with(self::PARAM_RESOURCE)
-            ->willReturn(null);
+        $exception = new Exception('test');
 
         $this->resourceRepository
             ->expects($this->once())
-            ->with($resourceRepositoryContext)
-            ->method('delete');
+            ->method('delete')
+            ->willThrowException($exception);
```
[the remark this answers is not in the document]
Comment (wazelin) [reply test/unit/core/resource/Service/ResourceDeleterTest.php:117]: Why not?

[the remark this answers is not in the document]
Comment (kilatib) [reply test/unit/core/resource/Service/ResourceDeleterTest.php:117]: oh sorry my mistake I read it as `without` somehow

**Review by kilatib — APPROVED** [review]

File: composer.json [hunk 5]
```diff
@@ -57,13 +57,13 @@
...
     "easyrdf/easyrdf": "^1.1",
-    "doctrine/dbal": "2.10.1",
+    "doctrine/dbal": "^2.10.1",
     "doctrine/annotations": "~1.6.0",
     "laminas/laminas-servicemanager": "~2.5.0",
     "league/flysystem": "~1.0",
     "league/flysystem-memory": "~1.0",
     "oat-sa/oatbox-extension-installer": "~1.1||dev-master",
-    "oat-sa/lib-generis-search": "2.1.2",
+    "oat-sa/lib-generis-search": "^2.1.2",
```
Comment (poyuki) [inline composer.json:66]: if i remember correctly this versioning means >= 2.1.2 && <2.1.3, is it initial idea for of this change? is this restriction make sense?

Comment (wazelin) [reply composer.json:66]: I'm pretty sure it means `>=2.1.2 <3.0.0-0`.

Comment (wazelin) [reply composer.json:66]: https://jubianchi.github.io/semver-check/#/^2.1.2/2.9.9

**Pull request merged**